Patch 1

steled opened the pull request:
<!-- 
Thanks for creating this pull request 🤗

Please make sure that the pull request is limited to one type (docs, feature, etc.) and keep it as small as possible. You can open multiple prs instead of opening a huge one.
-->

## 📑 Description
<!-- Add a brief description of the pr -->
This PR adds the possibility to update the deployment strategy.
As per defaul tit is set to rolling update but I'm facing problems with it and at least I want to use the strategy `Recreate`.

<!-- You can also choose to add a list of changes and if they have been completed or not by using the markdown to-do list syntax
- [ ] Not Completed
- [x] Completed
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information
<!-- Any additional information like breaking changes, dependencies added, screenshots, comparisons between new and old behavior, etc. -->
Before this PR it was not possible to set the strategy.
Now it can be set in the values file via:

```yaml
  strategy:
    type: Recreate
```

pmoscode commented:
Hi, thanks for your contribution.
Could you add a test for this change? Either in "tests/deployment_default_test.yaml" or a new one.

steled commented:
Hi, I added a new test file.
As I'm new to Helm unittest I just copied [tests/deployment_default_test.yaml](https://github.com/pmoscode-helm/jdownloader2/blob/main/tests/deployment_default_test.yaml) and added only these [3 lines](https://github.com/pmoscode-helm/jdownloader2/pull/58/files#diff-be649631a6db918b5da21bfe0d96b066300a542c74b53e8aeae1996943bb6d91R40-R42).

pmoscode commented:
Ok, now it's passing the test, but you didn't test the default behavior, when strategy is not set. When not set, no additional content/fields should be added when Helm has rendered the template.
And you don't need to include the default tests again. So, this test could look like this: (This test will fail, when strategy is not set, because just an empty "strategy"-field is rendered)

```
suite: test strategy deployment
chart:
  version: 9.9.9+test
  appVersion: 9.9.9
tests:
  - it: should not render strategy by default
    template: templates/deployment.yaml
    asserts:
      - notExists:
          path: spec.strategy

  - it: should render correct strategy settings (Recreate)
    template: templates/deployment.yaml
    set:
      strategy.type: Recreate
    asserts:
      - exists:
          path: spec.strategy.type
      - equal:
          path: spec.strategy.type
          value: Recreate
      - notExists:
          path: spec.strategy.rollingUpdate.maxSurge
      - notExists:
          path: spec.strategy.rollingUpdate.maxUnavailable

  - it: should render correct strategy settings (RollingUpdate)
    template: templates/deployment.yaml
    set:
      strategy.type: RollingUpdate
      strategy.rollingUpdate.maxSurge: 2
      strategy.rollingUpdate.maxUnavailable: 1
    asserts:
      - exists:
          path: spec.strategy.type
      - equal:
          path: spec.strategy.type
          value: RollingUpdate
      - exists:
          path: spec.strategy.rollingUpdate.maxSurge
      - equal:
          path: spec.strategy.rollingUpdate.maxSurge
          value: 2
      - exists:
          path: spec.strategy.rollingUpdate.maxUnavailable
      - equal:
          path: spec.strategy.rollingUpdate.maxUnavailable
          value: 1
```

Or you can set "RollingUpdate" as default value for "type" (Because this is the default for the "Deployment" resource. But then you need to adjust the tests.

steled commented:
Good morning, thanks for the input. I updated the deploment template to use an if statement, hope that this works correct now.

pmoscode commented:
Thanks, now, it looks good!
